mdsuite/transformations/velocity_from_positions.py

```diff
@@ -0,0 +1,59 @@
+"""
+This program and the accompanying materials are made available under the terms of the
+Eclipse Public License v2.0 which accompanies this distribution, and is available at
+https://www.eclipse.org/legal/epl-v20.html
+
+SPDX-License-Identifier: EPL-2.0
+
+Copyright Contributors to the MDSuite Project.
+
+Summary
+-------
+Calculate the velocity of particles from their positions
+"""
+import typing
+
+import tensorflow as tf
+
+from mdsuite.database.simulation_data_class import mdsuite_properties
+from mdsuite.transformations.transformations import SingleSpeciesTrafo
+
+
+class VelocityFromPositions(SingleSpeciesTrafo):
+    """
+    Calculate the velocity based on the particle positions via simple forward derivative,
+    i.e. v(t) = (x(t+dt)-x(t))/dt.
+    The last velocity of the trajectory cannot be computed and is copied
```

Comment:
I think if you do not have 'instant' velocity data and need to compute it from the positions, you need to have a fine enough resolution to capture all acceleration/deceleration of particles for this transformation to make sense. And if that resolution is good enough, autocorrelations should not suffer from a single slightly wrong value

Reply:
That sounds reasonable but perhaps this could be kept in mind. I just see for an autocorrelation function that for the last two time steps:

v(t) * v(t + 1) == 1,

which is probably just off of what it should be. However, as you mentioned for correct resolution of steps this should not cause any serious issue.